tests/scene/test_o_r_m_material_3d.h

```diff
@@ -0,0 +1,144 @@
+/**************************************************************************/
+/*  test_o_r_m_material_3d.h                                              */
+/**************************************************************************/
+/*                         This file is part of:                          */
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#pragma once
+
+#include "scene/resources/material.h"
+
+#include "tests/test_macros.h"
+
+namespace TestORMMaterial3D {
+
+TEST_CASE("[Material][ORMMaterial3D] Constructor & default state") {
+	if (!ClassDB::class_exists("ORMMaterial3D")) {
+		WARN_MESSAGE(true, "ORMMaterial3D is not available in this build.");
+	} else {
```

Comment:
I don't think we should have this check in every test case. Moreover, we should just skip running this test suite entirely in builds that don't have BaseMaterial3D. You should be able to achieve this by surrounding all test cases with `#ifndef _3D_DISABLED`.

Reply:
I tend to advise the use of Guard Clauses to reduce nesting and such https://wiki.c2.com/?GuardClause